### PR TITLE
fishPlugins.hydro: 2022-02-21 → 2024-03-24

### PR DESCRIPTION
**Pull request by spiwocoal**
## Description of changes

Solves #327417 and https://github.com/jorgebucaran/hydro/issues/55.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/shells/fish/plugins/hydro.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
-  version = "unstable-2022-02-21";
+  version = "unstable-2024-03-24";
 
   src = fetchFromGitHub {
     owner = "jorgebucaran";
     repo = "hydro";
-    rev = "d4875065ceea226f58ead97dd9b2417937344d6e";
-    sha256 = "sha256-nXeDnqqOuZyrqGTPEQtYlFvrFvy1bZVMF4CA37b0lsE=";
+    rev = "bc31a5ebc687afbfb13f599c9d1cc105040437e1";
+    sha256 = "sha256-0MMiM0NRbjZPJLAMDXb+Frgm+du80XpAviPqkwoHjDA=";
```
Comment (pyrox0) [inline pkgs/shells/fish/plugins/hydro.nix:11]: Since this is using the SRI hash, please change it to `hash = "sha256-whatever";`

**Review by pyrox0 — CHANGES_REQUESTED** [review]
Aside from this one nit, looks good. If you wanted to remove the `with lib;` in the meta section, feel free to, but not necessary.

**Comment by spiwocoal** [comment]
I'm pretty new with Nix, what would removing `with lib;` accomplish?

**Comment by pyrox0** [comment]
> I'm pretty new to Nix, what would removing `with lib;` accomplish?

Nixing that suggestion, not needed. If you want more info on why not to use `with lib;`, see #208242 and https://nix.dev/guides/best-practices#with-scopes. everything else looks good though, so no need to merge that, it's good to merge as-is.

**Review by pyrox0 — APPROVED** [review]

**Pull request merged**